examples/contrastive/contrastive.cpp

```diff
@@ -0,0 +1,223 @@
+#include "common.h"
+#include "llama.h"
+
+#include <cmath>
+#include <cstdio>
+#include <string>
+#include <vector>
+#include <limits>
+
+int main(int argc, char ** argv) {
+    gpt_params params_expert;
+    gpt_params params_amateur;
+    if (argc == 1 || argv[1][0] == '-') {
+        printf("usage: %s EXPERT_MODEL_PATH AMATEUR_MODEL_PATH [PROMPT]\n" , argv[0]);
+        return 1;
+    }
+
+    if (argc >= 2) {
+        params_expert.model = argv[1];
+    }
+
+    if (argc >= 3) {
+        params_amateur.model = argv[2];
+    }
+
+    if (argc >= 4) {
+        params_expert.prompt = argv[3];
+        params_amateur.prompt = argv[3];
+    }
+
+    if (params_expert.prompt.empty()) {
+        params_expert.prompt = "Hello my name is";
+        params_amateur.prompt = "Hello my name is";
+    }
+
+    // total length of the sequence including the prompt
+    const int n_len = 32;
+
+    // init LLM
+
+    llama_backend_init(params_expert.numa);
+
+    // initialize the model
+
+    llama_model_params model_params = llama_model_default_params();
+
+    // model_params.n_gpu_layers = 99; // offload all layers to the GPU
+
+    llama_model * model_expert = llama_load_model_from_file(params_expert.model.c_str(), model_params);
+    llama_model * model_amateur = llama_load_model_from_file(params_amateur.model.c_str(), model_params);
+
+
+    if (model_expert == NULL) {
+        fprintf(stderr , "%s: error: unable to load expert model\n" , __func__);
+        return 1;
+    }
+
+    if (model_amateur == NULL) {
+        fprintf(stderr , "%s: error: unable to load amateur model\n" , __func__);
+        return 1;
+    }
+
+    // initialize the context
+
+    llama_context_params ctx_params = llama_context_default_params();
+
+    ctx_params.seed  = 1234;
```

Comment:
I know this pattern is used in other examples, but is there a reason _this one_ needs to use a fixed seed? If so, put a comment near this line to explain why. If not, remove this line to let llama.cpp use the current time as a seed.

Reply:
I've erased the line.